js/client-teambuilder.js

```diff
@@ -1699,19 +1699,25 @@
 
 			if (template.speciesid === 'meowstic') {
 				smogdexid = 'meowstic-m';
-			} else if (smogdexid === 'rotom' || smogdexid === 'deoxys' || smogdexid === 'kyurem' || smogdexid === 'giratina' || smogdexid === 'shaymin' || smogdexid === 'tornadus' || smogdexid === 'thundurus' || smogdexid === 'landorus' || smogdexid === 'pumpkaboo' || smogdexid === 'gourgeist' || smogdexid === 'arceus' || smogdexid === 'meowstic' || smogdexid === 'hoopa') {
-				if (template.forme) smogdexid += '-' + toId(template.forme);
+			} else if (template.forme) {
+				if (template.speciesid === 'vivillon-fancy' || template.speciesid === 'vivillon-pokeball') smogdexid = 'vivillon';
+				if (template.speciesid === 'keldeo-resolute') smogdexid = 'keldeo';
+				if (template.speciesid === 'basculin-blue-striped') smogdexid = 'basculin';
+				if (template.speciesid === 'greninja-ash') smogdexid = 'greninja';
+				if (template.speciesid === 'pikachu-belle' || template.speciesid === 'pikachu-phd' || template.speciesid === 'pikachu-pop-star' || template.speciesid === 'pikachu-rock-star' || template.speciesid === 'pikachu-libre' || template.speciesid === 'pikachu-cosplay') smogdexid = 'pikachu';
+				if (template.speciesid === 'castform-sunny' || template.speciesid === 'castform-snowy' || template.speciesid === 'castform-rainy') smogdexid = 'castform';
+				smogdexid += '-' + toId(template.forme);
```

Comment:
what...?

can you say what's wrong instead of just saying um? the reason this is here is because it checks through everything with a forme and the exceptions are above to prevent the link from redirecting the user to a non-operational page.